datalad/interface/run.py

```diff
@@ -0,0 +1,268 @@
+            if not runinfo:
+                yield dict(
+                    err_info, status='impossible',
+                    message='cannot re-run command, last saved state does not look like a recorded command run')
+                return
+            rec_msg, runinfo = runinfo.groups()
+            if message is None:
+                # re-use commit message, if nothing new was given
+                message = rec_msg
+            try:
+                runinfo = json.loads(runinfo)
+            except Exception as e:
+                yield dict(
+                    err_info, status='error',
+                    message=('cannot re-run command, command specification is not valid JSON: %s', e.message))
+                return
+            if 'cmd' not in runinfo:
+                yield dict(
+                    err_info, status='error',
+                    message='cannot re-run command, command specification missing in recorded state')
+                return
+            cmd = runinfo['cmd']
+            rec_exitcode = runinfo.get('exit', 0)
+            rel_pwd = runinfo.get('pwd', None)
+            if rel_pwd:
+                # recording is relative to the dataset
+                pwd = normpath(opj(ds.path, rel_pwd))
+            else:
+                rel_pwd = None  # normalize, just in case
+                pwd = None
+
+            # now we have to find out what was modified during the last run, and enable re-modification
+            # ideally, we would bring back the entire state of the tree with #1424, but we limit ourself
+            # to file addition/not-in-place-modification for now
+            to_unlock = []
+            for r in ds.diff(
+                    recursive=True,
+                    revision='HEAD~1...HEAD',
+                    return_type='generator',
+                    result_renderer=None):
+                if r.get('type', None) == 'file' and \
+                        r.get('state', None) in ('added', 'modified'):
+                    r.pop('status', None)
+                    to_unlock.append(r)
+            if to_unlock:
+                for r in ds.unlock(to_unlock, return_type='generator', result_xfm=None):
+                    yield r
+        else:
+            # not a rerun, figure out where we are running
+            pwd = getpwd()
+            rel_pwd = relpath(pwd, start=ds.path)
+            if rel_pwd.startswith(pardir):
+                lgr.warning('Process working directory not inside the dataset, command run may not be reproducible.')
+                rel_pwd = None
+
+        # anticipate quoted compound shell commands
+        cmd = cmd[0] if isinstance(cmd, list) and len(cmd) == 1 else cmd
+
+        # TODO do our best to guess which files to unlock based on the command string
+        #      in many cases this will be impossible (but see --rerun). however,
+        #      generating new data (common case) will be just fine already
```

Comment:
mental note to myself -- apparently `sed -i` operates on a copy and then just replaces original one, so on my simple attempts it just worked without explicit unlocking